### PR TITLE
[flow] upgrade from 0.43 to 0.46

### PR DESCRIPTION
**Pull request by arthur801031**
Associated Issue: #2647 

### Summary of Changes
* Upgrade flow from 0.43 to 0.46
* Fix flow errors

### Test Plan
Pass
* yarn run flow
* yarn run lint
* yarn run test

### CODE REVIEW

**Comment by jasonLaster** [comment]
Thanks @arthur801031! I think you need to include the `yarn.lock`

Also - flow is now on 46 if that's alright :)

**Comment by arthur801031** [comment]
Hi @jasonLaster !

Done. Please check. Thank you! :)

File: src/utils/editor/source-search.js [hunk 1]
```diff
@@ -201,13 +201,18 @@ function searchNext(ctx, rev, query, newQuery, modifiers) {
...
+    }
+
     let cursor = getSearchCursor(cm, state.query, pos, modifiers);
 
     const location = rev
       ? { line: cm.lastLine(), ch: null }
       : { line: cm.firstLine(), ch: 0 };
 
-    if (!cursor.find(rev)) {
+    if (!cursor.find(rev) && state.query) {
```
Comment (AnshulMalik) [inline src/utils/editor/source-search.js]: Umm, If we have added a condition above to check state.query, then state.query will always evaluate to true at this point  isn't it ?

Comment (arthur801031) [inline src/utils/editor/source-search.js]: Yes. That's what I thought as well, but if we don't include this here, we'll get
screen shot 2017-05-21 at 1 33 02 pm

Comment (AnshulMalik) [inline src/utils/editor/source-search.js]: Oh. Okay

Comment (jasonLaster) [inline src/utils/editor/source-search.js]: Yep this makes sense. I bet we can better type state and not have these issues :)

**Comment by jasonLaster** [comment]
Thanks @arthur801031. staying on tip Flow has been a huge benefits

**Pull request merged**